### PR TITLE
Implement comparison operator for `StrainCollection` class

### LINKED ISSUE #110: Implement comparison operator for `StrainCollection` class
Currently there is no way how to compare two strain collections - we need a comprehensive way to check whether two `StrainCollection` objects are actually identical or not.

### PR DESCRIPTION
**Pull request by hechth**
Implemented operators for comparison of strains and straincollections based on attributes.
Added related tests.

Fixes #110

### CODE REVIEW

File: src/nplinker/strains.py [hunk 1]
```diff
@@ -38,3 +38,8 @@ def __repr__(self) -> str:
 
     def __str__(self) -> str:
         return f'Strain({self.id}) [{len(self.aliases)} aliases]'
+    
+    def __eq__(self, other):
+        result = self.id == other.id
+        result &= self.aliases == other.aliases
+        return result
```
Comment (CunliangGeng) [inline src/nplinker/strains.py]:
```suggestion
        return (
            isinstance(other, Strain)
            and self.id == other.id
            and self.aliases == other.aliases
        )
```

File: src/nplinker/strain_collection.py [hunk 2]
```diff
@@ -210,3 +210,9 @@ def __str__(self):
 
         return f'StrainCollection(n={len(self)}) [' + ','.join(
             s.id for s in self._strains) + ']'
+    
+    def __eq__(self, other):
+        result = self._strains == other._strains
```
Comment (CunliangGeng) [inline src/nplinker/strain_collection.py:215]: the `self._strains` has the type `list[Strain]`. so the order of Strain objects in the list also matters for the comparison. Do we care about the order? Does it make more sense to change the type to `set[Strain]`?

**Review by CunliangGeng — CHANGES_REQUESTED** [review]
Thanks, see the following comments

File: src/nplinker/strain_collection.py (hunk 2 above)
Comment (hechth) [reply src/nplinker/strain_collection.py:215]: I guess we shouldn't care about the order - so lets create an issue to see how it works when changing the type to `set[Strain]`, but let's keep this separate.

**Review by CunliangGeng — APPROVED** [review]

**Pull request merged**